### PR TITLE
EDU-1028 Support range in plyr player

### PR DESCRIPTION
**Pull request by adrianaluchian**
(Thank the Norse Gods it) closes https://educandu.atlassian.net/browse/EDU-1028

### CODE REVIEW

File: src/components/media-player/plyr/youtube-player.js [hunk 1]
```diff
@@ -0,0 +1,318 @@
+import Plyr from 'plyr';
+import PropTypes from 'prop-types';
+import { useStableCallback } from '../../../ui/hooks.js';
+import PlayIcon from '../../icons/media-player/play-icon.js';
+import { memoAndTransformProps } from '../../../ui/react-helper.js';
+import { useMediaDurations, useYoutubeThumbnailUrl } from '../media-hooks.js';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { MEDIA_ASPECT_RATIO, MEDIA_PROGRESS_INTERVAL_IN_MILLISECONDS } from '../../../domain/constants.js';
+
+export const YOUTUBE_STATE = {
```
Comment (ahelmberger) [inline src/components/media-player/plyr/youtube-player.js:10]: I do not see any outside usage, so I think we can drop the `export`

Comment (adrianaluchian) [reply src/components/media-player/plyr/youtube-player.js]: Indeed, dropped!

File: src/components/media-player/plyr/youtube-player.js [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+      player.volume = volume;
+    }
+  }, [player, volume]);
+
+  const handleEnded = useCallback(() => {
+    onEnded();
+    setProgressInterval(null);
+    setPlaybackEnded(true);
+    // compensate for cases where youtube actual source duration is shorter than reported,
+    // thus having playback end 1s earlier; likely a sounding issue on their side
```
Comment (ahelmberger) [inline src/components/media-player/plyr/youtube-player.js:174]: Or even likelier a **r**ounding issue (although it has sound)

Comment (adrianaluchian) [reply src/components/media-player/plyr/youtube-player.js]: Such are the wa[ys]/[ves] of the brain 😂 Corrected, thanks!

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**